### ISSUE TITLE
Make ggsave return filename

### ISSUE THREAD

**Issue by eliocamp** [tidyverse/ggplot2#3379]
`ggsave()` now returns `invisible()` (i.e. `NULL`). I think having it (invisibly) return the complete path to the saved image would be useful.

**Comment by HenrikBengtsson** [comment]
Came here with the same feature request; being able to do:
```r
pathname <- ggsave(filename, path = path, ...)
```
would be handy.

**Comment by yutannihilation** [comment]
Thanks for catching this. I also felt this was convenient and nice (and was almost approving the PR), but I'm not sure what (and whether or not) tidyverse has some rule on this kind of functions. For example, `readr::write_csv()` returns the data, not the filename.

``` r
x <- readr::write_csv(data.frame(x = 1:3), tempfile())
x
#>   x
#> 1 1
#> 2 2
#> 3 3
```

<sup>Created on 2020-03-11 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

But you example seems convincing.

**Comment by yutannihilation** [comment]
Closed by #4396

**Comment by hesscl** [comment]
This change breaks an unintended usage of `ggsave()` where users `+` it to the `ggplot()` they have assembled. It's discussed here [here](https://www.google.com/search?q=ggsave&oq=ggsave&aqs=chrome..69i57j69i60l2j69i61.2382j0j1&sourceid=chrome&ie=UTF-8) on StackOverflow and I thought I would mention it here since it appeared no one had yet.

The discussion here suggested some ambivalence about needing `ggsave()` to return filepaths, so to the extent that it is breaking some folks code I wondered if reverting might be worth consideration.
